src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -124,12 +126,30 @@ public void launch(SlaveComputer computer, TaskListener listener) {
             pod = client.pods().inNamespace(namespace).create(pod);
             LOGGER.log(INFO, "Created Pod: {0}/{1}", new Object[] { namespace, podId });
             listener.getLogger().printf("Created Pod: %s/%s%n", namespace, podId);
+
+            TaskListener runListener = template.getListener();
+            runListener.getLogger().printf("Created Pod: %s in namespace %s%n", podId, namespace);
+
+
             String podName = pod.getMetadata().getName();
             String namespace1 = pod.getMetadata().getNamespace();
             template.getWorkspaceVolume().createVolume(client, pod.getMetadata());
-            watcher = new AllContainersRunningPodWatcher(client, pod);
+            watcher = new AllContainersRunningPodWatcher(client, pod, runListener);
             try (Watch _w = client.pods().inNamespace(namespace1).withName(podName).watch(watcher)) {
                 watcher.await(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+            } catch (InvalidDockerImageException e) {
+                if (e.getMessage().contains("Unable to pull Docker image")) {
```

Comment:
Updated. Just an FYI this could become useful in the future if there are different error messages for different reasons why a Docker image can't be pulled. Have not seen this case yet though so removing for now.